tmt/convert.py

```diff
@@ -513,18 +522,36 @@ def read(
         beaker_task, data = \
             read_datafile(path, filename, datafile, types, testinfo)
 
-        # Warn if makefile has extra lines in run and build targets
-        def target_content(target: str) -> List[str]:
-            """ Extract lines from the target content """
-            regexp = rf"^{target}:.*\n((?:\t[^\n]*\n?)*)"
+        # Warn if makefile has extra lines in run target
```

Comment:
Why  not allow this multiline split into all targets? It is a feature of Makefile - https://www.gnu.org/software/make/manual/html_node/Splitting-Lines.html

IMO no need to make run special. User wanted to split lines in Makefile -> user gets it.

Reply:
Hi @lukaszachy, for `run`: user gets multiple lines and saves it in a single string, and the existing [target_content](https://github.com/teemtee/tmt/blob/main/tmt/convert.py#L510) return a list of lines. So I made run special. Without making it special, any other suggestion?

Reply:
Hi @lukaszachy, I debugged the related code block.  If we keep using function `target_content()`, it only get one line of the target `run` via regex `r"^run:.*\n((?:\t[^\n]*\n?)*)"`, e.g.

```python
(Pdb) b tmt/convert.py:530
Breakpoint 1 at /home/huanli/.virtualenvs/tmt/lib/python3.11/site-packages/tmt/convert.py:530
(Pdb) c
Checking the '/home/huanli/dev/1844/foo1' directory.
Makefile found in '/home/huanli/dev/1844/foo1/Makefile'.
task: /kernel/ipmixxx/foo
summary: Test XXX
test: ( set +o posix; . /usr/bin/rhts_environment.sh; \
. /usr/share/beakerlib/beakerlib.sh; \
. runtest.sh )
framework: shell
contact: Rachel Sibley <rasibley@redhat.com>
component: ipmitool
duration: 5m
recommend: beakerlib beakerlib-redhat
> /home/huanli/.virtualenvs/tmt/lib/python3.11/site-packages/tmt/convert.py(530)read()
-> if '\\\n' in datafile:
(Pdb) l
525  	                return []
526  	            target = search_result.group(1)
527  	            return [line.strip('\t') for line in target.splitlines()]
528  	
529  	        run_target_list = target_content("run")
530 B->	        if '\\\n' in datafile:
531  	            pass
532  	        else:
533  	            run_target_list.remove(data["test"])
534  	        if run_target_list:
535  	            echo(style(
(Pdb) p run_target_list
['( set +o posix; . /usr/bin/rhts_environment.sh; \\']
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ [1]
(Pdb) pp data["test"]
('( set +o posix; . /usr/bin/rhts_environment.sh; \\\n'
 '. /usr/share/beakerlib/beakerlib.sh; \\\n'
 '. runtest.sh )')
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ [2]
```

Right here we cannot use `run_target_list.remove(data["test"])` to make sure `data["test"]`  is the expected target, any suggestion?

Reply:
Fair enough. OK.